main.py

```
@@ -0,0 +1,149 @@
+    async def close(self):
+        """Cleanup when bot shuts down"""
+        logging.info("🔌 Bot shutting down...")
+        if self.session:
+            await self.session.close()
+        await self.db.close()
+        await super().close()
+
+    async def on_message(self, message):
+        if not message.author.bot:
+            print(f"Received message: {message.content} from {message.author}")
+            await bot.process_commands(message)
```

Pylint (3.3.7)

[error] Possibly using variable 'bot' before assignment

(E0606)